src/games/supported/Freeway.cpp

```diff
@@ -16,7 +16,9 @@
 
 FreewaySettings::FreewaySettings() {
 
-    reset();
+    m_reward   = 0;

```

Comment:
IF we keep the reset(System, Environment) function (see below):

We shouldn't duplicate code, especially because a few roms with have non-trivial values (e.g. Pitfall! sets score to 2000, and lives data also). A few possibilities:

- Keep the existing reset() method, and call the new one reset_and_set_mode()
- add a reset_variables() method that wraps the setting of values
- OR, See below for a preferred solution